src/Native/System.Native/pal_io.cpp

```diff
@@ -443,3 +464,93 @@ int32_t MksTemps(char* pathTemplate, int32_t suffixLength)
 {
     return mkstemps(pathTemplate, suffixLength);
 }
+
+static
+void ConvertPalPollFdToNative(PollFd& pal, pollfd& native)
+{
+    native.fd = pal.FileDescriptor;
+    native.events = pal.EventsToPollFor;
+    native.revents = pal.ReceivedEvents;
+}
+
+static
+void ConvertNativePollfdToPal(pollfd& native, PollFd& pal)
+{
+    pal.FileDescriptor = native.fd;
+    pal.EventsToPollFor = native.events;
+    pal.ReceivedEvents = native.revents;
+}
+
+extern "C"
+int32_t Poll(PollFd* pollData, uint32_t numberOfPollFds, int32_t timeout)
+{
+    // Convert our standardized pollfd to the native one
+    pollfd* fds = new pollfd[numberOfPollFds];
```

Comment:
If I greped correctly, we only call this with up to 2 fds. Seems like a waste to heap alloc/delete. Could we just put a stackalloc limit and fail otherwise?